src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs

```diff
@@ -13,7 +13,7 @@ public class MenuShapes : IShapeTableProvider
         public void Discover(ShapeTableBuilder builder)
         {
             builder.Describe("Menu")
-                .OnProcessing(async context =>
+                .OnDisplaying(async context =>
```

Comment:
But this will be called even if the shape is cached, isn't it? That's why I made OnProcessing, such that even if a shape is displayed, we only compute some data if it is not retrieved from the cache.

Reply:
Ah okay, i will try it.

Hmm, if i revert it to `OnProcessing()`, maybe we could still use the other change so that, as i have tried, a `Menu-MainMenu` template file and a `Menu__MainMenu` database template would work, which is not the case currently. Note: without thinking here that it is only related to `Menu` shapes.